tools/toolchains/__init__.py

```diff
@@ -1231,9 +1231,25 @@ def mem_stats(self, map):
 
         return None
 
+    def add_regions(self):
+        """Add regions to the build profile, if there are any.
+        """
+        print("Using regions in this build:")
+        for region in self.config.regions:
+            for define in [(region.name.upper() + "_ADDR", region.start),
+                           (region.name.upper() + "_SIZE", region.size)]:
+                self.cc.append("-D%s=0x%x" %  define)
+                self.cppc.append("-D%s=0x%x" %  define)
+                if region.active:
+                    self.ld.append(self.make_ld_define(*define))
```

Comment:
Will this still work for `MBED_APP_START`? It seems like `MBED_APP_START_ADDR` would be the new define?

Reply:
I just tried these changes with [bootloader blinky](https://github.com/armmbed/mbed-os-example-bootloader-blinky) for K64F and it did not work. I could see that it only `-DAPPLICATION_ADDR` and `-DAPPLICATION_SIZE` were passed to the linker. The k64f linker script expects `MBED_APP_START`.

Reply:
Yep. Let me fix that.